### ISSUE TITLE
Cannot use "pytest test_*" (with wildchar filter) to collect cases on Windows 7 command line

### ISSUE THREAD

**Issue by chenzx** [pytest-dev/pytest#6971]
"pytest test_*" cmd lets me selectively run specific tests which are classified by filename prefixes
this works on my mac perfectly
but when i try the trick on a windows 7 machines, pytest cannot collect any cases.

**Comment by RonnyPfannschmidt** [comment]
at first glance i strongly suspect this is a shell issue
in mac its likely that bash (i hope thats correct, i use linux)just expands `test_*` to a number of filenames

im not aware about how the shells on windows manage it,
my understanding is, that on windows, the programs are supposed to expand wildcards, which pytest currently does not do

adding passed wild-card expansion would be a new feature,
if only support foor `*` expansion is added,
i believe it wouldn't be a breaking change

in that case its a help wanted item

**Comment by The-Compiler** [comment]
Pretty much what @RonnyPfannschmidt says - also see e.g. https://github.com/psf/black/issues/1198 and https://github.com/pallets/click/issues/1096.

Note that if pytest did that kind of expansion, it should be done on Windows only.

**Comment by RonnyPfannschmidt** [comment]
i believe support on all platforms is possibly sensible, as it would allow wildcard support in `PYTEST_ADDOPTS` and the `addopts` option

**Comment by Zac-HD** [comment]
The problem with expanding wildcards is that it's possible to have a file literally named `test_*` that you want to run pytest on, and then what?

**Comment by The-Compiler** [comment]
> The problem with expanding wildcards is that it's possible to have a file literally named `test_*` that you want to run pytest on, and then what?

That isn't possible on Windows - hence my proposal to only do the expansion on Windows. Yes, that means there's no way to use wildcards in `PYTEST_ADDOPTS` and `addopts`, but I can't think of a good reason to use them there to be honest.

**Comment by RonnyPfannschmidt** [comment]
Good point, we should have a option to control the behaviour with sane defaults for posix vs windows

**Comment by harishkesavarao** [comment]
Has this been fixed via #1830?

**Comment by RonnyPfannschmidt** [comment]
The linked issue it's about raises?
